modules/payloads/singles/linux/x86/shell_bind_tcp_super_small_random_port.rb

```diff
@@ -0,0 +1,55 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+module MetasploitModule
+
+  CachedSize = 44
+
+  include Msf::Payload::Single
+  include Msf::Payload::Linux
+
+  def initialize(info = {})
+    super(merge_info(info,
+      'Name'          => 'Linux Command Shell, Bind TCP Random Port Inline',
+      'Description'   => %q{
+        The tiniest (46 bytes!) bind tcp shell in its class! Listen for a connection with a random port and spawn a command shell.
+        Use nmap to discover the open port: 'nmap -sS -p- target'.
```

Comment:
```suggestion
        Use nmap to discover unfiltered closed port: 'nmap -sS -p- target'.
```

with BIND TCP, you need an available, not filtered, closed port, not an open one.

Reply:
phra, yes, we need not filtered and not used (closed) port when it comes to bind tcp.

But, once shellcode triggers - it opens a port, so we need to scan a target to find that opened port in order to connect to it.

Reply:
got it. :+1: